include/revng/Model/Types.h

```diff
@@ -0,0 +1,335 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <algorithm>
+#include <optional>
+
+#include "llvm/ADT/StringRef.h"
+
+#include "revng/ADT/SortedVector.h"
+#include "revng/Support/Assert.h"
+#include "revng/Support/Debug.h"
+
+namespace flatc {
+
+struct Type {
+  enum TypeKind {
+    TK_Invalid,
+    TK_Primitive,
+    TK_Enum,
+    TK_Typedef,
+    TK_Struct,
+    TK_Union,
+    TK_FunctionPtr,
+  };
+
+  std::string Name; // Key
+
+private:
+  TypeKind Kind;
+
+public:
+  TypeKind getKind() const { return Kind; }
```

Comment:
Please follow the conventions in `Binary.h`, no private fields, no accessors.

Reply:
It's already like that in the latest changes I just pushed.